static/js/hotkey.js

```diff
@@ -216,7 +216,7 @@ export function in_content_editable_widget(e) {
 
 // Returns true if we handled it, false if the browser should.
 export function process_escape_key(e) {
-    if (hashchange.in_recent_topics_hash() && recent_topics.change_focused_element(e, "escape")) {
+    if (recent_topics.is_visible() && recent_topics.change_focused_element(e, "escape")) {
```

Comment:
What are the semantics of this when an overlay like settings is present on top of "recent topics"?  I think `recent_topics.is_visible` is false in that case?

In any case, should this block sit below most of the rest of this function, including the `processing_text` part?